### PR TITLE
Remove SYNCHMGR_PIPE_BASED_THREAD_BLOCKING and clean up synchmanager

### PR DESCRIPTION
**Pull request by adityamandaleeka**
This change deletes a bunch of code under the macro SYNCHMGR_PIPE_BASED_THREAD_BLOCKING which is not enabled anywhere.

Additionally, while touching the synchmanager code recently, I noticed a bunch of small issues to clean up such as removing goto's, fixing typos, etc. I've done those as part of this change as well.

There shouldn't be any functional difference with this change, but I did notice a potential existing bug while cleaning up goto's. There is an impossible condition at  https://github.com/dotnet/coreclr/blob/597f60b93facd540ca18fe6bf7aa3bb022f3da8b/src/pal/src/synchmgr/synchmanager.cpp#L1184 

Basically we'll never add to the state controller cache here. Perhaps the intention in that 'if' statement was to check for `NO_ERROR == palErr` instead of `NO_ERROR != palErr`? If that's the case, I'll fix that with my change. Otherwise, we should delete that block.

cc: @jkotas

### CODE REVIEW

**Comment by janvorli** [comment]
LGTM

**Comment by adityamandaleeka** [comment]
@dotnet-bot test this please

**Comment by adityamandaleeka** [comment]
@dotnet-bot test this please
